### PR TITLE
Add networking config guide.

### PR DESCRIPTION
**Pull request by avaidyanatha**
# Main Changes
- Adds the networking guide written by @michaeljguarino with some updates for devX.

### CODE REVIEW

File: advanced-topics/network-configuration.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+{% endtab %}
+{% endtabs %}
+
+{% hint style="warning" %}
+💡 Note that updating these will likely cause the VPC to be replaced, which will recreate your cluster. We recommend that you destroy your cluster before applying network config modifications to it.
+{% endhint %}
+
+## Configuring VPC Peering
+
+Plural creates a fresh VPC and Kubernetes cluster for deploying your applications into. This ensures that we have a clean environment to deploy into and minimizes disruption to existing systems. It does come with the tradeoff of slightly increased network complexity, but most cloud providers can bridge this using VPC peering. This is a technology that effectively allows you to combine the address spaces of two VPCs in the cloud you operate in.
```
Comment (michaeljguarino) [inline advanced-topics/network-configuration.md:52]: Probably should rm the trailing into on the first sentence, or reword it somehow.

File: advanced-topics/network-configuration.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+resource "aws_vpc_peering_connection" "foo" {
+   peer_owner_id = "my-owner-id" # Use appropriate values here.
+   peer_vpc_id   = "peer-vpc-id"
+   vpc_id        = module.aws-bootstrap.vpc.id
+}
+```
+{% endtab %}
+
+{% tab title="GCP" %}
+```bash
```
Comment (michaeljguarino) [inline advanced-topics/network-configuration.md:70]: I think we might be able to use terraform as a codeblock in gitbook (notion didn't support it)

File: advanced-topics/network-configuration.md (hunk 1 above)
Comment (avaidyanatha) [reply advanced-topics/network-configuration.md]: Yep, makes sense - fixed.

File: advanced-topics/network-configuration.md (hunk 2 above)
Comment (avaidyanatha) [reply advanced-topics/network-configuration.md:70]: I checked, it wasn't available unfortunately.

Comment (michaeljguarino) [reply advanced-topics/network-configuration.md:70]: is hcl available?

Comment (michaeljguarino) [reply advanced-topics/network-configuration.md:70]: I think it is, we should use that, it's basically the same

**Pull request merged**